store/auth.go

```diff
@@ -159,3 +160,25 @@ func DischargeAuthCaveat(username, password, macaroon, otp string) (string, erro
 	}
 	return responseData.Macaroon, nil
 }
+
+// authError contains the reason behind an authentication failure
+type authError struct {
+	Threshold int64  `json:"threshold"`
+	Error     string `json:"error"`
+}
+
+// decodeUnauthorizedError attempts to determine the cause of an authentication failure
+// against the purchasing server.
+func decodeUnauthorizedError(body io.ReadCloser) error {
+	var msg authError
+	dec := json.NewDecoder(body)
+	if err := dec.Decode(&msg); err != nil {
+		return fmt.Errorf("store authorization failed with unknown error: %v", err)
```

Comment:
this message sound strange, shouldn't it say that it couldn't json decode it, if we captured the body fuflly before we could print at least instead, othewise it sounds like err is the error we got